gh-128588: gh-128550: remove eager tasks optimization that missed and introduced incorrect cancellations (#129063)
python/cpython: Misc/NEWS.d/next/Library/2025-01-20-13-12-39.gh-issue-128550.AJ5TOL.rst

> Fixed an incorrect optimization relating to eager tasks in :class:`asyncio.TaskGroup` that resulted in cancellations being missed.

Author: ```suggestion
Removed an incorrect optimization relating to eager tasks in :class:`asyncio.TaskGroup` that resulted in cancellations being missed.
```

> remove a problematic optimization relating to eager tasks in :class:`asyncio.TaskGroup` that resulted in cancellations being missed

Reviewer: ```suggestion
Fixed an incorrect optimization relating to eager tasks in :class:`asyncio.TaskGroup` that resulted in cancellations being missed.
```
